rotkehlchen/api/v1/schemas.py

```diff
@@ -1170,6 +1175,30 @@ def _validate_blockchain_account_schemas(
                 )
             given_addresses.add(address)
 
+    # Make sure bitcoin cash addresses are valid
+    elif data['blockchain'] == SupportedBlockchain.BITCOIN_CASH:
+        for account_data in data['accounts']:
+            address = address_getter(account_data)
+            # Check whether its a valid BCH CashAddr
+            if address.startswith('bitcoincash:') and not is_valid_bitcoin_cash_address(address):
+                raise ValidationError(
+                    f'Given value {address} is not a valid bitcoin cash address',
+                    field_name='address',
+                )
+            # Check to see whether its a valid BCH legacy address
+            if (not address.startswith('bitcoincash:') and address.endswith('.eth')) and not is_valid_btc_address(address):  # noqa: 501
```

Comment:
Can't you combine all these 3 if checks into another function and move it outside of api/v1/schemas? Probably somewhere in bitcoin/utils.py?